src/fiesta/train/AfterglowData.py

```
@@ -670,6 +675,158 @@ def __call__(self, idx):
+
+        X, y = self.run_afterglow_model(X_raw)
+        return X, y
+
+    def run_afterglow_model(self, X):
+        y = np.empty((len(X), len(self.nus), len(self.times)))
+        runner = RunBlastwaveRS(self.times, self.nus, X, self.parameter_names, self.fixed_parameters)
+        for idx in tqdm.tqdm(range(len(X)), desc=f"Computing {len(X)} blastwave+RS calculations.", leave=False):
+            try:
+                i, out = runner(idx)
+                y[i] = out
+            except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)